src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -1002,6 +1002,71 @@ internal static char ParseFormatSpecifier(ReadOnlySpan<char> format, out int dig
             }
         }
 
+        private static string? FormatBigIntegerToBin(BigInteger value, int digits)
+        {
+            // Get the bytes that make up the BigInteger.
+            byte[]? arrayToReturnToPool = null;
+            Span<byte> bits = stackalloc byte[64]; // arbitrary threshold
+            if (!value.TryWriteOrCountBytes(bits, out int bytesWrittenOrNeeded))
+            {
+                bits = arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bytesWrittenOrNeeded);
+                bool success = value.TryWriteBytes(bits, out bytesWrittenOrNeeded);
+                Debug.Assert(success);
+            }
+            bits = bits.Slice(0, bytesWrittenOrNeeded);
+
+            Debug.Assert(!bits.IsEmpty);
+
+            byte highByte = bits[bits.Length - 1];
+            int charCount = highByte == 0 ? 1 : 8 - byte.LeadingZeroCount(highByte);
+            charCount += (bits.Length - 1) * 8;
+            if (digits > charCount)
+            {
+                charCount = digits;
+            }
+
+            // each byte is typically eight chars
+            var sb = new ValueStringBuilder(stackalloc char[Math.Min(charCount, 512)]);
```

Comment:
maybe
```csharp
var sb = (charCount <= 512)
   ? new ValueStringBuilder(stackalloc char[charCount])
   : new ValueStringBuilder(charCount);
}
```

Reply:
Fixed, didn't know this pattern can mark `sb` as 'unsafe to return' for stackalloc before, thanks !